static/js/admin.js

```diff
@@ -875,6 +904,50 @@ function _setup_page() {
         });
     });
 
+    $("#alias_list").on("click", ".delete_alias", function (e) {
+        var url = "/json/realm/aliases/" + e.currentTarget.id;
+        var row = $(this).closest('tr');
+
+        // Clear previous error.
+        $("#realm_aliases_modal").find(".aliases_error").text("");
+
+        channel.del({
+            url: url,
+            success: function () {
+                var index = page_params.domains.indexOf(row.find('.domain').text());
+                page_params.domains.splice(index, 1);
+                update_alias_label();
+                row.remove();
+            },
+            error: function (xhr) {
+                $("#realm_aliases_modal").find(".aliases_error").text(JSON.parse(xhr.responseText).msg);
+            }
+        });
+    });
+
+    $("#add_alias").click(function () {
+        var data = {
+            domain: $("#new_alias").val(),
+        };
+
+        // Clear previous error.
+        $("#realm_aliases_modal").find(".aliases_error").text("");
+
+        channel.post({
+            url: "/json/realm/aliases",
+            data: data,
+            success: function (response_data) {
+                add_alias(response_data.new_alias);
+                page_params.domains.push(response_data.new_alias[1]);
+                update_alias_label();
+                $("#new_alias").val("");
```

Comment:
You'll need to rearrange this javascript a bit once you use the proper real-time sync approach; see below.